cmd/account/mgmt/account-assign.go

```diff
@@ -0,0 +1,185 @@
+	return &accountAssignOptions{
+		flags:      flags,
+		printFlags: printer.NewPrintFlags(),
+		IOStreams:  streams,
+	}
+}
+
+// assignCmd assigns an aws account to user under osd-staging-2 by default unless osd-staging-1 is specified
+func newCmdAccountAssign(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	ops := newAccountAssignOptions(streams, flags)
+	accountAssignCmd := &cobra.Command{
+		Use:               "assign",
+		Short:             "Assign account to user",
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	ops.printFlags.AddFlags(accountAssignCmd)
+	accountAssignCmd.Flags().StringVarP(&ops.payerAccount, "payer-account", "p", "", "Payer account type")
+	accountAssignCmd.Flags().StringVarP(&ops.username, "username", "u", "", "LDAP username")
+
+	return accountAssignCmd
+}
+
+func (o *accountAssignOptions) complete(cmd *cobra.Command, _ []string) error {
+	if o.username == "" {
+		return cmdutil.UsageErrorf(cmd, "LDAP username was not provided")
+	}
+	if o.payerAccount == "" {
+		return cmdutil.UsageErrorf(cmd, "Payer account was not provided")
+	}
+
+	return nil
+}
+
+func (o *accountAssignOptions) run() error {
+
+	var (
+		accountAssignID string
+		destinationOU   string
+		rootID          string
+	)
+
+	rootID = OSDStaging2RootID
+	destinationOU = OSDStaging2OuID
+	if o.username != "" && o.payerAccount == "osd-staging-1" {
+		rootID = OSDStaging1RootID
+		destinationOU = OSDStaging1OuID
+	}
+	//Instantiate aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+
+	o.awsClient = awsClient
+	accountAssignID, err = o.findUntaggedAccount()
+	if err != nil {
+		return err
+	}
+
+	err = o.tagAccount(accountAssignID)
+	if err != nil {
+		return err
+	}
+
+	err = o.moveAccount(accountAssignID, destinationOU, rootID)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintln(o.IOStreams.Out, accountAssignID)
+
+	return nil
+}
+
+var ErrNoUntaggedAccounts = fmt.Errorf("no untagged accounts available")
+var ErrNoAccountsInRoot = fmt.Errorf("no accounts available in root")
+
+func (o *accountAssignOptions) findUntaggedAccount() (string, error) {
+	//List accounts that are not in any OU
+	input := &organizations.ListAccountsInput{}
+	accounts, err := o.awsClient.ListAccounts(input)
+	if err != nil {
+		return "", err
+	}
+	if len(accounts.Accounts) == 0 {
+		return "", ErrNoAccountsInRoot
+	}
+
+	// Loop through accounts and check that it's untagged and assign ID to user
+	var accountAssignID string
+	for _, a := range accounts.Accounts {
+
+		inputListTags := &organizations.ListTagsForResourceInput{
+			ResourceId: a.Id,
+		}
+		tags, err := o.awsClient.ListTagsForResource(inputListTags)
+		if err != nil {
+			return "", err
+		}
+		if len(tags.Tags) == 0 {
```

Comment:
I'm not sure if we want to specifically look at the length of the tags slice, this might put us in a situation where we tag accounts for other reasons and this starts failing, even if those accounts are eligible to be claimed.

We should at least check for the `claimed: true` tag.